### PR TITLE
Moving to go 1.17

### PR DESCRIPTION
**Pull request by IgorGov**
Why so many indirect modules were added?
[Go 1.17 release notes](https://go.dev/doc/go1.17) 

> Because the number of explicit requirements may be substantially larger in an expanded Go 1.17 go.mod file, the newly-added requirements on indirect dependencies in a go 1.17 module are maintained in a separate require block from the block containing direct dependencies.

### CODE REVIEW

**Comment by mertyildiran** [comment]
I guess this image requires an upgrade as well https://github.com/up9inc/mizu/blob/0365b8971f6bc4743d7bf223ede668f91dd257fe/devops/linux-arm64-musl-go-libpcap/Dockerfile#L4-L5

**Comment by IgorGov** [comment]
> I guess this image requires an upgrade as well

Good catch! Thanks @mertyildiran

**Comment by mertyildiran** [comment]
> > I guess this image requires an upgrade as well
> 
> Good catch! Thanks @mertyildiran

That image also requires a build and push on an ARM64 machine after the merge.

**Comment by IgorGov** [comment]
> That image also requires a build and push on an ARM64 machine after the merge.

done

File: tap/extensions/amqp/go.mod [hunk 1]
```diff
@@ -1,7 +1,9 @@
 module github.com/up9inc/mizu/tap/extensions/amqp
 
-go 1.16
+go 1.17
 
 require github.com/up9inc/mizu/tap/api v0.0.0
 
+require github.com/google/martian v2.1.0+incompatible // indirect
```
Comment (gadotroee) [inline tap/extensions/amqp/go.mod:7]: why does the AMQP has reference to martian?

Comment (gadotroee) [reply tap/extensions/amqp/go.mod:7]: I guess it is because the "tap/api" module has maritan?

Comment (IgorGov) [reply tap/extensions/amqp/go.mod:7]: yep

**Review by gadotroee — APPROVED** [review]

**Pull request merged**